### ISSUE TITLE
Containers don't update their dns (/etc/resolv.conf) through DHCP.

### ISSUE THREAD

**Issue by AJNOURI** [docker/docker#11695]
**Description of problem:**
I use one container as a DHCP server to assign IP and DNS inf. to other containers.
- DHCP client containers do not change their DNS inf. (/etc/resolv.conf)
- Cannot requery new DHCP from interactive console

**docker version:**
Docker version 1.5.0, build a8a31ef

Containers: 6
Images: 45
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 57
Execution Driver: native-0.2
Kernel Version: 3.13.0-46-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 4
Total Memory: 7.798 GiB
Name: hostname
ID: ...:3BTV
Username: ajnouri
Registry: [https://index.docker.io/v1/]
WARNING: No swap limit support

**uname -a:**
Linux hostname 3.13.0-46-generic #79-Ubuntu SMP Tue Mar 10 20:06:50 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty

**Environment details:**
Physical machine

**Actual Results:**
Client container get assigned an ip address, but not dns informations (*/etc/resolv.conf* still holds old inf. 8.8.8.8). 
And all attempts to release and get reassigned ip failed from the container:

*root @ f1de9e5b5528:# sudo dhclient eth1 -r*
```Failed to send flush request: Operation not permitted```

*root @ f1de9e5b5528:# sudo dhclient eth1*
```RTNETLINK answers: Operation not permitted```
```RTNETLINK answers: Operation not permitted```
```mv: cannot move '/etc/resolv.conf.dhclient-new.188' to '/etc/resolv.conf': Device or resource busy```

**Expected Results:**
- Client container get assigned an ip address through DHCP.
- Client change their dns server using DHCP (/etc/resolv.conf).

**Comment by estesp** [comment]
Hi there--what you are trying to do conflicts with the default network support implemented via the Docker daemon bridge (`docker0`) and container `veth` devices.  You might want to take a look at @jpetazzo 's `pipework` tool (https://github.com/jpetazzo/pipework) and also note that Docker is working on a pluggable network implementation so that specialized container network implementations can be selected vs. the default `docker0` bridge + daemon-assigned IP strategy.

In short, you would have to specify `--net=none` on the `docker run` line for containers as a first step to trying to have DHCP assigned addresses; `pipework` docs will explain this as well.

**Comment by AJNOURI** [comment]
Hi estesp, I am experiencing the same problem with pipework:
- Containers communicating through a new interface (other than eth0), still DHCP client containers receive and set their IP according to DHCP server container, but the same error in observed with /resolv.conf.

**Comment by unclejack** [comment]
@AJNOURI If you want to do something like this, you need to use `--net=none` as @estesp has stated above and set up your custom network using pipework.

Such custom setups aren't supported right now, but once the implementation for #9983 lands, it should be easier to have such custom setups.

In order to get step by step help and discuss similar questions, please post this question and similar questions on the docker-user Google group (https://groups.google.com/forum/#!forum/docker-user) or on the forums (https://forums.docker.com/). There are others out there who have done something similar and those two public forums are recommended for such discussions.

I'm going to close this issue since it's about a custom use case which is going to be addressed through the resolution of other issues. Please feel free to comment.
